docs/deploys.md

```diff
@@ -0,0 +1,120 @@
+
+1. **Configuration**: The script generated by Halyard has commands to copy all
+ configuration from the necesssary Spinnaker config already generated by 
+ Halyard into `/opt/spinnaker/config`, `/opt/spinnaker-monitoring` and 
+ `/etc/apache2`. Any required files are already in the correct location.
+2. **Service Dependency Setup**: The only service dependency is Redis, and it
+ is installed locally.
+3. **Deployment & Updates**: All spinnaker packages are pinned by writing
+ `apt` preferences in `/etc/apt/preferences.d/pin-spin-{artifact}` for each
+ Spinnaker artifact. During the initial install and subsequent updates, this
+ file is always updated before attempting to grab new packages via `apt`.
+
+### LocalhostGit Deployments
+
+> Timeline: Mid-to-late April
+
+Halyard handles this deployment by installing all Spinnaker service git repos
+in a user-configurable directory, with a user-configurable `upstream` and
+`origin`.  There will be a special version called `HEAD` that supersedes any 
+configuration in our profile registry that will take all services at git `HEAD` 
+to use for config generation.
+
+### Flotilla Deployments
+
+> Timeline: Varies by cloud provider. See below.
+
+1. **Configuration**: Varies by cloud provider.
+2. **Service Dependency Setup**: Varies by cloud provider.
+3. **Deployent & Updates**: This deployment involves first deploying a separate 
+ Orca, Clouddriver and Redis using their published `-bootstrap.yml` profiles to 
+ only communicate with one-another, and then orchestrate the remainder of the 
+ update/deployment through them by publishing red/black update pipelines that 
+ redeploy desired services with a cluster history size of 2 for fast rollbacks.
+ There are a few nuances here:
+  * **Orca**: Orca must have an unlimited cluster history size, and Halyard
+   must decide when it is safe to kill an Orca node.
+  * **Cache-schema changes**: Cache changes have broken service updates
+   the past, so we need to handle these separately. Since there will be window 
+   when instances from both version with incompatable caches schemas will 
+   share the same Redis cache, flushing the cache is not an option. The only 
+   alternative is to scale down the old version to 0 instances, and the flush
+   the redis cache where applicable. The critical thing to note here is that
+   the only way Halyard can know to do that is when a services Major version
+   increases.
+
+#### Kubernetes 
+
+> Timeline: End of March. The core features are in, but a few touchups as
+> detailed in https://github.com/spinnaker/halyard/projects/1
+
+1. **Configuration**: Halyard publishes all configuration as Kubernetes
+ secrets, and describes to the individual services where to mount them.
+2. **Service Dependency Setup**: The only external dependency is redis, and if
+ it is not provided, Spinnaker will deploy a Redis instance.
+3. **Deployent & Updates**: See above.
+
+#### GCE
+
```

Comment:
Right, it's not mentioned but I'm assuming the agents are installed on the nodes as a part of our bake process, and it's just a matter of providing them with config.